### PR TITLE
[LibraryImportGenerator] Reduce unnecessary casting/locals in pinning path

### PR DESCRIPTION
**Pull request by elinor-fung**
Contributes to https://github.com/dotnet/runtime/issues/69608: Style item (6)

UTF-16 `string`:
```csharp
[LibraryImport("lib", StringMarshalling = StringMarshalling.Utf16)]
internal static partial void ManyStrings(string s1, string s2, string s3, string s4, string s5);

// Generated
internal static partial void ManyStrings(string s1, string s2, string s3, string s4, string s5)
{
    // Pin - Pin data in preparation for calling the P/Invoke.
    fixed (void* __s1_native = s1)
        fixed (void* __s2_native = s2)
            fixed (void* __s3_native = s3)
                fixed (void* __s4_native = s4)
                    fixed (void* __s5_native = s5)
                    {
                        __PInvoke(__s1_native, __s2_native, __s3_native, __s4_native, __s5_native);
                    }

    // Local P/Invoke
    [System.Runtime.InteropServices.DllImportAttribute("lib", EntryPoint = "ManyStrings", ExactSpelling = true)]
    static extern unsafe void __PInvoke(void* s1, void* s2, void* s3, void* s4, void* s5);
}
```

User-defined: 
https://github.com/dotnet/runtime/blob/d143e18e1ac2894b47dad9fe062dbe1244e0f611/src/libraries/System.Runtime.InteropServices/tests/TestAssets/SharedTypes/NonBlittable.cs#L98-L104
```csharp
[LibraryImport("lib")]
public static partial void UserDefinedPinnable(SharedTypes.IntWrapper i1, SharedTypes.IntWrapper i2);

// Generated
public static partial void UserDefinedPinnable(global::SharedTypes.IntWrapper i1, global::SharedTypes.IntWrapper i2)
{
    // Pin - Pin data in preparation for calling the P/Invoke.
    fixed (void* __i1_native = i1)
        fixed (void* __i2_native = i2)
        {
            __PInvoke((int*)__i1_native, (int*)__i2_native);
        }

    // Local P/Invoke
    [System.Runtime.InteropServices.DllImportAttribute("lib", EntryPoint = "UserDefinedPinnable", ExactSpelling = true)]
    static extern unsafe void __PInvoke(int* i1, int* i2);
}
```

### CODE REVIEW

**Comment by am11** [comment]
Do we have such control over indentation which makes it (conveniently) generate stacked `fixed` statements like this:

```c#
    // Pin - Pin data in preparation for calling the P/Invoke.
    fixed (void* __s1_native = s1)
    fixed (void* __s2_native = s2)
    fixed (void* __s3_native = s3)
    fixed (void* __s4_native = s4)
    fixed (void* __s5_native = s5)
    {
        __PInvoke(__s1_native, __s2_native, __s3_native, __s4_native, __s5_native);
    }
```

File: src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/MarshallingGenerator.cs [hunk 1]
```diff
@@ -56,7 +56,11 @@ public enum ValueBoundaryBehavior
         /// <summary>
         /// The address of the native identifier provided by <see cref="StubCodeContext.GetIdentifiers(TypePositionInfo)"/> should be passed by value.
         /// </summary>
-        AddressOfNativeIdentifier
+        AddressOfNativeIdentifier,
```
Comment (AaronRobinsonMSFT) [inline src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/MarshallingGenerator.cs:59]:
```suggestion
        AddressOfNativeIdentifier,
        
```

Comment (AaronRobinsonMSFT) [reply src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/MarshallingGenerator.cs:59]: Same above? Missing line between value and comment.

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Comment by elinor-fung** [comment]
We are relying on Roslyn's [`NormalizeWhitespace`](https://docs.microsoft.com/dotnet/api/microsoft.codeanalysis.csharp.syntaxextensions.normalizewhitespace) to format / apply indentation accordingly to the entire block of generated code, so we don't fully control the indentation like that. I opened https://github.com/dotnet/roslyn/issues/61518 to track potentially improving the behaviour around cases like this.

File: src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/MarshallingGenerator.cs (hunk 1 above)
Comment (elinor-fung) [reply src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/MarshallingGenerator.cs:63]: Added throughout.

**Comment by elinor-fung** [comment]
Failures are https://github.com/dotnet/runtime/issues/69792.

**Pull request merged**